### DISCUSSION TITLE
Debug Indicatons

### DISCUSSION THREAD

**Discussion by siddhc** [flandreas/antares#501] [Q&A]
On running a simulation:
____________________
After some clock ticks Error 1 occurred (please see this screenshot):
![Error Stage 1 of 2](https://user-images.githubusercontent.com/18248762/198897205-49209b39-223a-4e6b-af5a-898047bfd1e5.png)
My question no. one: I _alt-clicked_ the sub-circuits to open them but found no red lines. At this stage what more I can do to get a better direction towards the bug in my circuit?
____________________
I clicked _un-pause_ and the simulation moved forward. After some clock ticks Error 1 happened again. After some more clock ticks Error 2 occurred (please see this screenshot):
![Error Stage 2 of 2](https://user-images.githubusercontent.com/18248762/198897365-1e47746f-a8b4-4d02-ab3d-88028ce5ba79.png)
My question no. two: Great to see a more detailed indication, but how do I figure out the parts (11) and (5) i.e. "Tri-state buffer (11)" and "8BRwOC ... (5)"?
____________________
Thanks for your patience to go through the query. Request your thoughts and indications.

**Comment by flandreas** [comment]
@siddhc
Question No. 1: If the simulation is paused due to an error, the "Issues" tab at the window's bottom edge should have a red icon and contain an error issue with a description of the cause.

Question No. 2: The number in parentheses in the error description (such as (11)) designates the "Model ID" of the corresponding component. You see the "Model ID" of a component in the "Properties" window when you select the component in edit mode. In your case, the tristate buffer with model ID 11 asserts the signal 1 to a wire, while a "8 Bit Register w O/P Control v0.4" with model ID 5 asserts the signal 0000ZZZZ to the same wire.

**Comment by flandreas** [comment]
@siddhc Have you been able to make progress in investigating the issue?

You could also export your project and attach it to this ticket. I could then perhaps use it to think about what features are missing in Antares that would help debugging such situations.

**Reply by siddhc** [reply]
@flandreas Thanks for your response. I could not find the "Issues" tab and the red icon. Please let me know where to locate it.
Not yet solved by me. The project export is here:
[8 Bit Computer 2022.10.31 23.57.zip](https://github.com/flandreas/antares/files/9903876/8.Bit.Computer.2022.10.31.23.57.zip)
--------------------------------------------------------------------------------------------------------
Also, two points (could be feature requests, in case I have missed them): 

_[Point 1 of 2]_
Say there are 8 red LEDs. now I want to change all of them to green. I cannot select all 8 and change them to green at once in properties. I need to do them separately.

_[Point 2 of 2]_
Similarly, say I have a tri-state buffer and an LED. I want to change the Orientation of both to North. I cannot select them together and change Orientation (or any other common property).

Request your view.

**Reply by siddhc** [reply]
Note: I do not know why is the first part above appearing in big fonts. No purpose, not able to reduce the font size.

**Reply by flandreas** [reply]
"Issues" tab: See encircled in red in the screenshot below. It would frighten me if you don't have that..
<img width="1347" alt="Screenshot 2022-10-31 at 19 35 03" src="https://user-images.githubusercontent.com/24633141/199084400-f2354c36-6188-4124-90c5-2e2848a67a98.png">

Okay, thanks for your project zip. I will look into it...

Your point 1: No, changing the same property in multiple components in one go is not supported, sorry. That would be an interesting feature request, but not trivial to implement.

Your point 2: You can select multiple components and then the press Cmd-R (or menu Edit->Rotate) to rotate them all in one go. But again, not via the properties window.

Never mind about the big font, not problem.

**Reply by flandreas** [reply]
@siddhc I've started the simulation of "8b Computer v0.3a", but wasn't able to produce a simulation error. Perhaps this is because I'm not sure which inputs I have to set at what moment, or because the project zip only contains "8b Computer" up to v0.3a, but your screenshot with the simulation error seems to be referring to v0.4.

**Reply by siddhc** [reply]
@flandreas Thanks for your reply.
... Did not reproduce because "which inputs when" was the reason.
... The error was due to a wrongly firing PC, it is sorted with a tristate buffer now.
... On the "Issues" tab on bottom left, I can figure it only now after you told me. See my comments in BLUE in this image for three situations. Also, I do not see the exclamation triangle next to 'Issues'. I am running version 1.1.0.
![Issues](https://user-images.githubusercontent.com/18248762/200182737-9f6f5520-4065-4ad2-a73e-86b7a2f082c7.png)

Above for information please.

**Reply by flandreas** [reply]
@siddhc The "Issues" tab is an expandable view, just like the "Explorer" tab on the left side of the Antares window. When you click on the "Issues" button, the view is displayed, and when you click again on the "Issues" button, the view is hidden.

Is it possible that the "29C Smoke" thing on your PC is hiding the lower part of the Antares window? If you make the Antares window smaller, you should see the "Issues" button and the "Log" button, which is also located at the bottom of the Antares window.

**Reply by flandreas** [reply]
@siddhc Point 1 is going to be implemented with #506 in the next release

**Reply by siddhc** [reply]
@flandreas 
"The "Issues" tab ... view is hidden.": Yes it is visible, was hidden behind the task bar. Sorted on this.

"Point 1 ... release": Superb !
